src/execution/operator/aggregate/physical_perfecthash_aggregate.cpp

```diff
@@ -99,16 +107,44 @@ void PhysicalPerfectHashAggregate::Sink(ExecutionContext &context, GlobalOperato
 		group_chunk.data[group_idx].Reference(input.data[bound_ref_expr.index]);
 	}
 	idx_t aggregate_input_idx = 0;
-	for (idx_t i = 0; i < aggregates.size(); i++) {
-		auto &aggr = (BoundAggregateExpression &)*aggregates[i];
+	for (auto & aggregate : aggregates) {
+		auto &aggr = (BoundAggregateExpression &)*aggregate;
 		for (auto &child_expr : aggr.children) {
 			D_ASSERT(child_expr->type == ExpressionType::BOUND_REF);
 			auto &bound_ref_expr = (BoundReferenceExpression &)*child_expr;
 			aggregate_input_chunk.data[aggregate_input_idx++].Reference(input.data[bound_ref_expr.index]);
 		}
+		if (aggr.filter) {
+			vector<LogicalType> types;
+			vector<vector<Expression *>> bound_refs;
+			BoundAggregateExpression::GetColumnRef(aggr.filter.get(), bound_refs, types);
```

Comment:
Same here; can't this just refer to the filter as computed in the projection above it?